### PR TITLE
Document using RabbitMQ for main contributors

### PR DESCRIPTION
**Pull request by yvanzo**
It provides guidelines on documenting and implementing the use of self-hosted RabbitMQ by MetaBrainz projects.
The goal is to make projects easier to deploy and to maintain, even by system admins.

### CODE REVIEW

**Comment by yvanzo** [comment]
For reference, the projects using RabbitMQ already have some bits of such documentation:
- CAA: https://github.com/metabrainz/CAA-indexer/blob/master/Architecture.md
- LB: https://listenbrainz.readthedocs.io/en/latest/developers/architecture/#services
- MB SIR: https://sir.readthedocs.io/en/stable/setup/index.html#amqp-setup

File: services/README.md
Comment (reosarevok) [inline services/README.md]: "of" seems wrong here, just "when using a self-hosted service" I guess? :)

File: services/README.md
Comment (reosarevok) [inline services/README.md]: depeds -> depends

**Review by reosarevok — DISMISSED** [review]
Other than two typos (marked) the wording here seems sensible to me and the RabbitMQ example seems useful. I'll leave it to others to confirm whether any extra details are missing.

Should the generic RabbitMQ file contain the links to the per-project docs to make them easier to find, or is it enough that the projects have them?

File: services/README.md
Comment (yvanzo) [reply services/README.md]: Correct, fixed in https://github.com/metabrainz/guidelines/pull/17/commits/e1d5fcb5665cc986dafb4357e47348cd5850c3f5

File: services/README.md
Comment (yvanzo) [reply services/README.md]: Thanks, fixed in https://github.com/metabrainz/guidelines/pull/17/commits/e1d5fcb5665cc986dafb4357e47348cd5850c3f5 too

**Comment by yvanzo** [comment]
> the RabbitMQ example seems useful. I'll leave it to others to confirm whether any extra details are missing.

The idea is to try to follow the RabbitMQ guidelines for all projects using it before writing guidelines for other services (Redis, Postgres…). It would allow to see if this approach works and if changes must be made before looking into other services.

> Should the generic RabbitMQ file contain the links to the per-project docs to make them easier to find, or is it enough that the projects have them?

Good suggestion. Originally I thought about linking to per-project docs from our private admin RabbitMQ doc. But centralizing these docs publicly seem to be more sensible. Added an “Examples” section and a link to related tickets in commit <https://github.com/metabrainz/guidelines/pull/17/commits/a5abcf4d68968f2b292af50028181680877c5bdc>.

Edit: “@yvanzo dismissed @reosarevok’s [stale review](https://github.com/metabrainz/guidelines/pull/17#pullrequestreview-985872969) via [e1d5fcb](https://github.com/metabrainz/guidelines/commit/e1d5fcb5665cc986dafb4357e47348cd5850c3f5)” 1 hour ago is automatic when updating the PR.

File: services/RabbitMQ.md
Comment (amCap1712) [inline services/RabbitMQ.md:1]: Since the title is general guidelines for RabbitMQ use, should we also add here to document things like acknowledgement mode ?

Comment (yvanzo) [reply services/RabbitMQ.md:1]: Sure. I don’t know what it is but the guidelines are here for that.

**Review by amCap1712 — DISMISSED** [review]

File: services/RabbitMQ.md
Comment (yvanzo) [reply services/RabbitMQ.md:1]: For the record, it has been added in commit https://github.com/metabrainz/guidelines/pull/17/commits/1df992397ea2dbef93917dd464cea84c60dca08e.

**Comment by yvanzo** [comment]
Rebased to squash amending commits. (Review has been dismissed automatically.)

**Review by alastair — APPROVED** [review]
The readme file looks good. Based on this we can start to work on docs for timescale, typesense, and redis for LB.

I'm not as familiar with the rabbitmq side of LB, so can't comment too much on the specifics in that document, but this looks like a good start for the rmq documentation

**Pull request merged**
